src/taco-utils/promisesUtils.ts

```diff
@@ -0,0 +1,31 @@
+﻿/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../typings/lodash.d.ts" />
+
+"use strict";
+
+import _ = require("lodash");
+
+module TacoUtility {
+    export class PromisesUtils {
+        public static executeAfter<T>(valueOrPromise: T, afterCallback: { (): void }, failCallback?: { (reason: any): void }): T {
```

Comment:
I don't think that's quite right, or at least not useful then:

If you are computing something synchronously, then
`executeAfter(someSyncCalculation(), () => console.log("done"), () => console.log("failed"))` is equivalent to `var result = someSyncCalculation(); executeAfter(result, () => console.log("done"), () => console.log("failed"))` and if the result isn't a promise, then it's just `var result = someSyncCalculation(); console.log("done")`. I don't see why you would invoke it like this...